### PR TITLE
chore: don't unwrap

### PR DESCRIPTION
**Pull request by ctron**
Here's a proposal for getting rid of `.unwrap()`, which we use a bit to often for my taste.

I know it's controversial, and I think it's ok/good to keep them for test code. But not in the code we want to run in production. I also know that sometimes it's hard to work around that. But then let's isolate this, and be explicit about it.

This PR is a start, and if we agree we can do the rest before merging.

### CODE REVIEW

File: analytics/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![deny(clippy::unwrap_used)]
```
Comment (jcrossley3) [inline analytics/src/lib.rs:1]: I don't love this.

File: bombastic/api/src/server.rs [hunk 2]
```diff
@@ -73,15 +74,18 @@ enum Error {
     InvalidContentEncoding,
 }
 
+#[allow(clippy::unwrap_used)]
```
Comment (jcrossley3) [inline bombastic/api/src/server.rs:77]: I don't love this

File: bombastic/api/src/server.rs [hunk 3]
```diff
@@ -142,18 +146,16 @@ async fn query_sbom(
...
-        head.content_encoding.and_then(|ref e| {
-            accept_encoding
-                .negotiate([e.parse().unwrap()].iter())
-                .map(|s| s.to_string())
-                .filter(|x| x == e)
-        })
+        head.content_encoding
+            .as_ref()
+            .and_then(|e| e.parse::<Encoding>().ok())
+            .and_then(|e| accept_encoding.negotiate([&e].into_iter()).filter(|x| x == &e))
```
Comment (jcrossley3) [inline bombastic/api/src/server.rs:152]: I *DO* love this. I would've loved having this feedback when the PR was reviewed.

File: bombastic/api/src/server.rs [hunk 4]
```diff
@@ -218,17 +220,17 @@ impl From<&SearchParams> for SearchOptions {
...
+tag = "bombastic",
+path = "/api/v1/sbom/search",
+responses(
+(status = 200, description = "Search completed"),
+(status = BAD_REQUEST, description = "Bad query"),
+(status = 401, description = "Not authenticated"),
+),
+params(
+("q" = String, Query, description = "Search query"),
+)
```
Comment (jcrossley3) [inline bombastic/api/src/server.rs:233]: You don't like the indenting?

File: bombastic/walker/src/lib.rs [hunk 5]
```diff
@@ -98,14 +100,18 @@ impl Run {
                     let keys = self
                         .signing_key
                         .into_iter()
-                        .chain(self.devmode.then(|| Url::parse(DEVMODE_KEY).unwrap()))
+                        .chain(
+                            self.devmode
+                                .then(|| Url::parse(DEVMODE_KEY).context("failed to parse devmode key"))
+                                .transpose()?,
+                        )
```
Comment (jcrossley3) [inline bombastic/walker/src/lib.rs:107]: Can this even happen? I like how this guy puts it: https://stackoverflow.com/a/39478185

**Review by jcrossley3 — COMMENTED** [review]
I'm conflicted...

File: bombastic/api/src/server.rs (hunk 4 above)
Comment (ctron) [reply bombastic/api/src/server.rs:233]: Indeed, that's intellij messing things up sometimes. I will fix this.

File: bombastic/walker/src/lib.rs (hunk 5 above)
Comment (ctron) [reply bombastic/walker/src/lib.rs:107]: It should not. Then again, that's what engineers think about most of their bugs. The alternative would be to use `.expect("message")`. Which is basically `unwrap` but with a proper message.

File: bombastic/api/src/server.rs (hunk 2 above)
Comment (ctron) [reply bombastic/api/src/server.rs:77]: That's an example for when you need to opt-out. Which is ok sometimes I guess.

File: admin/src/upload.rs [hunk 6]
```diff
@@ -131,12 +132,19 @@ async fn upload(
     let headers: HeaderMap = headers
         .iter()
         .map(|s| {
-            let mut parts = s.splitn(2, ':');
-            let key = parts.next().unwrap().trim();
-            let value = parts.next().unwrap().trim();
-            (key.parse().unwrap(), value.parse().unwrap())
+            let Some((key, value)) = s.split_once(':') else {
```
Comment (jcrossley3) [inline admin/src/upload.rs:135]: New syntax? I haven't seen this.

File: bombastic/api/src/server.rs [hunk 7]
```diff
@@ -73,15 +75,18 @@ enum Error {
...
+}
+
 impl error::ResponseError for Error {
     fn error_response(&self) -> HttpResponse {
         let mut res = HttpResponse::build(self.status_code());
         res.insert_header(ContentType::plaintext());
         match self {
             Self::InvalidContentType => res.insert_header(Accept::json()),
-            Self::InvalidContentEncoding => res.insert_header(AcceptEncoding(
-                ACCEPT_ENCODINGS.iter().map(|s| s.parse().unwrap()).collect(),
```
Comment (jcrossley3) [inline bombastic/api/src/server.rs:83]: Personally, I'm ok with the `unwrap` here. It's a non-public const. To me, as someone familiar with HTTP headers and their string values, the code is more clear without dragging in the `QualityItem`, `Preference`, and `Encoding` abstractions only to add an `allow` directive.

In the interest of clarity, I'd probably `s/unwrap/expect("unexpected")/` just to appease clippy.

File: admin/src/upload.rs (hunk 6 above)
Comment (ctron) [reply admin/src/upload.rs:135]: kind of, learned about that just recently … https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html

File: bombastic/api/src/server.rs (hunk 7 above)
Comment (ctron) [reply bombastic/api/src/server.rs:83]: The idea was to isolate that part into its own function. Changed it back.

**Review by jcrossley3 — APPROVED** [review]
Some of these are huge improvements, thanks!

**Pull request merged**